### PR TITLE
#202 introduce error page and routing

### LINKED ISSUE #202: Change default route - add error page
Currently any route redirects to the explorer. As part of setting up the hosting I was thinking of a page not found page or general error page. This is something of a frivolity - and likely of little practical value. Treat accordingly.

## Expected Behaviour
Display a page that informs the user if they've entered an incorrect URL or clicked an invalid link.
Tell the user in simple language if an error has occurred and provide a way to return to the homepage.

## Current Behaviour
Always redirects to 'explorer' regards of URL entered.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #202 

## Description
Something of a Friday frivolity. Likely at no one will reach this page - but it does tidy up an inconsistency and is a bit of fun.
An invalid URL redirects to `/404` which displays this:

Screen Shot 2020-09-28 at 9 16 44 AM

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

File: unified-codes/apps/web/src/components/templates/ErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export interface ErrorPageProps {
+  code: string;
+}
+
+export type ErrorPage = React.FunctionComponent<ErrorPageProps>;
+
+export const ErrorPage: ErrorPage = ({ code }) => {
+  switch (code) {
+    case '404':
```
Comment (wlthomson) [inline unified-codes/apps/web/src/components/templates/ErrorPage.tsx:13]: I think the `switch` statement would be something to lift out to the parent (i.e. don't couple the component with a specific `404` error or whatever, can always make another component for another use-case?

**Review by wlthomson — APPROVED** [review]
LGTM!

As part of the redesign, could envisage having templates: `template/Main.tsx` and `template/Error.tsx`. Could then define this page as `pages/NotFound.tsx`:

```
<Route exact path="/404">
  <ErrorPage body={<NotFound/>}/>
</Route>
```

Where the error template would define whatever layout the error webpage will have and the page defines the main content of the error webpage. E.g. of main page with explorer for context (where `EntityExplorer` is a page like `NotFound`, i.e. `pages/EntityExplorer.tsx`):

```
<Route exact path="/explorer">
  <MainPage
    header={<PageHeader/>
    body={<EntityExplorer/>}
    footer={<PageFooter/>)
  />
</Route>
```

...not sure if these prop names are the best, suggestions welcome!

Then we'd be pretty much on track with atomic design! Yes, no?

**Comment by wlthomson** [comment]
@mark-prins Btw, best error page ever!

**Comment by katherine-sussol** [comment]
> @mark-prins Btw, best error page ever!

I agree! Where did you get the image?! It's amazing

**Review by katherine-sussol — APPROVED** [review]
😻 👍

**Pull request merged**